docs/src/main/paradox/jdbc.md

```diff
@@ -1,8 +1,246 @@
+
+When using `JdbcProjection.exactlyOnce`, the `JdbcSession` that is passed to the handler will be used to save the offset behind the scenes. Therefore, it's extremely important to disable auto-commit (eg: `setAutoCommit(false)`), otherwise the two operations won't participate on the same transaction.  
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #jdbc-session-imports #jdbc-session }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #jdbc-session-imports #jdbc-session }
+
+
+@@@ div { .group-java }
+An alternative Hibernate based implementation would look like this:
+
+@@snip [HibernateJdbcSession.java](/examples/src/test/java/jdocs/jdbc/HibernateJdbcSession.java) { #hibernate-session-imports #hibernate-session } 
+
+And a factory to create new instances:
+
+@@snip [HibernateSessionFactory.java](/examples/src/test/java/jdocs/jdbc/HibernateSessionFactory.java) { #hibernate-factory-imports #hibernate-factory }
+
+@@@
+
+When declaring a `JdbcProjection` you must provide a factory for the `JdbcSession`. The factory will be used to create new instances whenever needed.
+
+@@@ note
+It's highly recommended configuring it with a connection pool, for example [HikariCP](https://github.com/brettwooldridge/HikariCP).
+@@@
+
+## Blocking JDBC Dispatcher
+
+JDBC APIs are blocking by design, therefore Akka Projection JDBC will use a dedicated dispatcher to run all JDBC calls. It's important to configure the dispatcher to have the same size as the connection pool. 
+
+Each time an envelope is delivered to the handler one thread and one database connection will be used. If your connection pool is smaller than the number of threads, the thread can potentially block while waiting for the connection pool to provide a connection. 
+
+The dispatcher pool size can be configured through the `akka.projection.jdbc.blocking-jdbc-dispatcher.thread-pool-executor.fixed-pool-size` settings. See @ref:[configuration section](#configuration) below.
+
+@@@ note
+Most applications will use database connections to read data, for instance to read a projected model upon user request. This means that other parts of the application will be competing for a connection. It's recommend to configure a connection pool dedicated to the projections and use a different one in other parts of the application.  
+@@@
 
 ## exactly-once
 
+The offset is stored in the same transaction used for the user defined `handler`, which means exactly-once
+processing semantics if the projection is restarted from previously stored offset.
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #projection-imports #exactlyOnce }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #exactlyOnce }
 
```

Comment:
add
```
The @ref:[`ShoppingCartHandler` is shown below](#handler).
```

Reply:
Sure. 

Also missing in the Slick one. I will add there as well.